frontend/src/component/project/ProjectList/ProjectList.tsx

```diff
@@ -86,16 +95,31 @@ function resolveCreateButtonData(
     }
 }
 
-const ProjectCreationButton = () => {
+const useProjectLimit = (projectsLimit: number, projectCount: number) => {
+    const resourceLimitsEnabled = useUiFlag('resourceLimits');
+    const limitReached = resourceLimitsEnabled && projectCount >= projectsLimit;
+
+    return limitReached;
+};
+
+const ProjectCreationButton: FC<{ projectCount: number }> = ({
+    projectCount,
```

Comment:
Question regarding SWR and shared caches etc:

Is there any point in passing the `projectCount` here instead of using `useProjects` directly in the button component? If it's a shared cache, then it doesn't incur any extra costs, right? With all the other stuff this button is doing (getting search params, access data, etc), it feels strange that _that's_ the thing we pass in.

Reply:
If we move useProjects inside the button then the button is aware of all the details of all the projects while it only needs to know how many projects we have. I introduced this param to limit the surface area of the things the button needs to know about projects.

Reply:
Yeah. This is a good point, and I'm honestly not sure what we typically do. On one extreme, you have the Elm architecture, right, where everything is passed down as props from above. But with SWR and react hooks, it's more like everything is accessible everywhere, so we grab what we need when we need it.

So in that case, it makes sense to me that this button, which is a highly specialized component already, can just go ahead and grab the projects directly if it needs them.

On the other hand, this kind of rampant implicit reading does make me throw up in my mouth a bit because it goes against a lot of what I have been taught. Also makes it harder to test when you need to set up a bunch of server routes etc. 

So I'm happy to go either way, but I'm not convinced that leaving the project count alone as something we need to pass makes the most sense. But I'm also not opposed to it. I see your reasoning and am happy for you to leave it as is. I also got to hear your thoughts on the matter, so that's great 😄

Reply:
For the one off component like this one it probably does not matter that much. I'd be more worried if we shared this component in 20 different places then the shape of the API and the corresponding tradeoffs would matter a lot more.